client/allocdir/alloc_dir.go

```diff
@@ -343,6 +360,38 @@ type ReadCloserWrapper struct {
 	io.Closer
 }
 
+func (d *AllocDir) BlockUntilExists(path string, t *tomb.Tomb) error {
+	// Get the path relative to the alloc directory
+	p := filepath.Join(d.AllocDir, path)
+
+	var watcher watch.FileWatcher
+	if runtime.GOOS == "windows" {
+		// There are some deadlock issues with the inotify implementation on
+		// windows. Use polling watcher for now.
+		watcher = watch.NewPollingFileWatcher(p)
+	} else {
+		watcher = watch.NewInotifyFileWatcher(p)
+	}
+
+	return watcher.BlockUntilExists(t)
+}
+
+func (d *AllocDir) ChangeEvents(path string, curOffset int64, t *tomb.Tomb) (*watch.FileChanges, error) {
+	// Get the path relative to the alloc directory
+	p := filepath.Join(d.AllocDir, path)
+
+	var watcher watch.FileWatcher
+	if runtime.GOOS == "windows" {
```

Comment:
With my nose in and eye on portability, can this be a helper function wrapped in a build tag?